### ISSUE TITLE
Can this be used in ROS 2?

### ISSUE THREAD

**Issue by heiyumiao** [matthias-mayr/Cartesian-Impedance-Controller#10]
Or what modifications do I need to make if I want to use it in ROS2

**Comment by matthias-mayr** [comment]
Good question. I haven't looked into ROS 2 control yet. Since the control algorithms are abstracted in a base class, these do not need to be touched. The ROS 1 integration is already a wrapper around them, so it might be comparably easy to make this ROS 2 compliant.

If you have the time to look into it, you can take a look at the [cartesian_controllers](https://github.com/fzi-forschungszentrum-informatik/cartesian_controllers) library that already did the ROS 2 port.
A pull request would definitely appreciated. Unfortunately I would not have time to look into a ROS 2 port in the next months myself.

**Comment by Daan0103** [comment]
any update on this one?

**Comment by matthias-mayr** [comment]
I learned a bit more about ros2_control in the meantime and it should be relatively to port this package.
However, I am afraid that I will not have any time myself to do this before February.

If anyone is interested in doing this and to send in a pull request, that would be greatly appreciated.

**Comment by mhubii** [comment]
agreed, since the code is nicely structured, this shouldn't be too difficult. Might give this a go, but currently busy doing other things as well. For reference https://github.com/lbr-stack/lbr_fri_ros2_stack/issues/163

**Comment by mrceki** [comment]
Hi all! 
I’ve just pushed the ROS 2 port to my [forked repository.](https://github.com/mrceki/Cartesian-Impedance-Controller) However, it’s not ready for a pull request yet and hasn’t been tested on the real robot—only in simulation so far. In the meantime, I’m open to any feedback and recommendations.

**Comment by mrceki** [comment]
Hi @matthias-mayr, I would be happy to submit PR if you could create a 'ros2' branch. Also, I run the controller on Franka Research 3. But to be sure, how did you guys test the controller on the robot? I want to test it correctly :)
